asset/asset.go

```diff
@@ -55,6 +55,21 @@ const (
 // compressed, 33-byte form.
 type SerializedKey [33]byte
 
+// NewSerializedKeyFromBytes creates a new SerializedKey from the given byte
+// slice.
+func NewSerializedKeyFromBytes(key []byte) (SerializedKey, error) {
+	var zero SerializedKey
+
+	if len(key) != 33 {
```

Comment:
Since we have an error return value, I think we should check that the point is actually on the curve by parsing it as `btcec.PublicKey`.